src/main/java/training/busboard/Models/Bus.java

```diff
@@ -0,0 +1,40 @@
+package training.busboard.Models;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+@JsonIgnoreProperties(ignoreUnknown=true)
+public class Bus {
+
+    public int timeToStation;
+    public String destinationName;
+    public String stationName;
+    public String lineName;
+    public String platformName;
+
+    public Bus() {
+    }
+
+    public Bus(int timeToStation, String destinationName, String stationName, String lineName, String platformName) {

```

Comment:
You can configure Jackson to use this constructor, which will let you delete the parameterless constructor and make all the fields final.

You can do this by adding a `@JsonCreator` annotation to the constructor.

Unfortunately you also have to annotate every field in the constructor because Jackson can't check the names of the parameters at runtime: `public Bus(@JsonProperty("timeToStation") int timeToStation, @JsonProperty("destinationName") String destinationName ...`

You can go one step further and add the Java compiler `-parameters` flag so that Jackson _can_ read the names of the parameters at runtime: https://manosnikolaidis.wordpress.com/2015/08/25/jackson-without-annotations/ Feel free to give it a go if Jackson's verbosity is annoying you, but don't worry about it if not.